executor/src/witgen/machines/block_machine.rs

```diff
@@ -279,7 +279,13 @@ impl<'a, T: FieldElement> Machine<'a, T> for BlockMachine<'a, T> {
                  This might violate some internal constraints."
             );
         }
-        self.degree = compute_size_and_log(&self.name, self.data.len(), self.degree_range);
+        self.degree = compute_size_and_log(
+            &self.name,
+            // At this point, the still contains the dummy block, which will be removed below.
+            // Therefore, we subtract the block size here.
```

Comment:
This feels kinda hacky..

Reply:
Hmm, so I think the hacky part is that the data has a block that will be removed later, right? Which was the case before this PR?

I tried refactoring it, but it's kinda complicated. LMK if you have a suggestion.